FirebaseSessions/Sources/Settings.swift

```diff
@@ -0,0 +1,167 @@
+  private let fileManager: SettingsFileManager
+  private let appInfo: ApplicationInfoProtocol
+  private let lock = NSLock()
+  private var _settingsDictionary: [String: AnyObject]?
+  private var isCacheKeyExpired: Bool
+  struct CacheKey: Codable {
+    var createdAt: Date
+    var googleAppID: String
+    var appVersion: String
+  }
+
+  var settingsDictionary: [String: AnyObject]? {
+    objc_sync_enter(lock)
+    defer { objc_sync_exit(lock) }
+    return _settingsDictionary
+  }
+
+  var sessionsEnabled: Bool {
+    guard let enabled = settingsDictionary?["sessions_enabled"]?.boolValue else {
+      return true
+    }
+    return enabled
+  }
+
+  var samplingRate: Double {
+    guard let rate = settingsDictionary?["sampling_rate"]?.doubleValue else {
+      return 1
+    }
+    return rate
+  }
+
+  var sessionTimeout: TimeInterval {
+    guard let timeout = settingsDictionary?["session_timeout"]?.doubleValue else {
+      return 30 * 60
+    }
+    return timeout
+  }
+
+  var isCacheExpired: Bool {
+    guard settingsDictionary != nil else {
+      return true
+    }
+    return isCacheKeyExpired
+  }
+
+  private var cacheDurationSeconds: TimeInterval {
+    guard let duration = settingsDictionary?["cache_duration"]?.doubleValue else {
+      return Settings.cacheDurationSecondsDefault
+    }
+    return duration
+  }
+
+  init(fileManager: SettingsFileManager = SettingsFileManager(), appInfo: ApplicationInfoProtocol) {
+    self.fileManager = fileManager
+    isCacheKeyExpired = false
+    self.appInfo = appInfo
+  }
+
+  func loadCache(googleAppID: String, currentTime: Date) {
+    guard let cacheData = fileManager.data(contentsOf: fileManager.settingsCacheContentPath) else {
+      Logger.logDebug("[Sessions:Settings] No settings were cached")
+      return
+    }
+    guard let parsedDictionary = cacheData.dictionaryValue else {
+      removeCache()
+      return
+    }
+    do {
+      objc_sync_enter(lock)
+      defer { objc_sync_exit(lock) }
+      _settingsDictionary = parsedDictionary
+    }
+    guard let cacheKeyData = fileManager.data(contentsOf: fileManager.settingsCacheKeyPath),
+          let cacheKey = cacheKeyData.cacheKeyValue else {
+      Logger.logError("[Sessions:Settings] Could not load settings cache key")
+      removeCache()
+      return
+    }
+    guard cacheKey.googleAppID == googleAppID else {
+      Logger
+        .logDebug("[Sessions:Settings] Invalidating settings cache because Google App ID changed")
+      removeCache()
+      return
+    }
+    if currentTime.timeIntervalSince(cacheKey.createdAt) > cacheDurationSeconds {
+      Logger.logDebug("[Sessions:Settings] Settings TTL expired")
+      do {
+        objc_sync_enter(lock)
+        defer { objc_sync_exit(lock) }
+        isCacheKeyExpired = true
+      }
+    }
+    if appInfo.synthesizedVersion != cacheKey.appVersion {
+      Logger.logDebug("[Sessions:Settings] Settings expired because app version changed")
```

Comment:
Kindof a nit - I believe we can leave off the `[Sessions:Settings]` and either have no prefix, or just use `[Settings]` as the prefix. Our logger by default adds a prefix to logs, eg:

```
 10.2.0 - [FirebaseSessions][I-SES000000] Logging events using fast QOS due to running on a simulator
```

Reply:
Done